test/jdk/java/util/regex/RegExTest.java

```diff
@@ -4987,26 +4267,25 @@ private static void expoBacktracking() throws Exception {
             String p = (String)pm[0];
             String s = (String)pm[1];
             boolean r = (Boolean)pm[2];
-            if (r != Pattern.compile(p).matcher(s).matches()) {
-                failCount++;
-            }
+            assertSame(r, Pattern.compile(p).matcher(s).matches());
```

Comment:
`expectThrows(PatternSyntaxException.class, ...)` might suit better for this and similar cases in this file. Not only does `expectThrows` test for expected exception, it also returns an exception which you can further inspect. Now, if we only had `assertThat` or similar functionality for compound assertions, the complete test might look nicely.

Reply:
The latter comment referred to hand-rolled try-fail constructs like this: https://github.com/openjdk/jdk/blob/3937bf8f9921395d6fcbdc2bb00a4a98dc2aaf62/test/jdk/java/util/regex/RegExTest.java#L4281-L4289

Not sure why my IDE plugin messed up the comment position.